### PR TITLE
Bug 1404147 - Prompt error message when create secret with invalid email

### PR DESCRIPTION
**Pull request by jhadvig**
@jwforres PTAL

### CODE REVIEW

File: app/views/directives/create-secret.html [hunk 1]
```diff
@@ -234,6 +234,11 @@
             required>
         </div>
       </div>
+      <div class="has-error" ng-show="secretForm.dockerEmail.$invalid && !secretForm.dockerEmail.$pristine">
+        <span class="help-block">
+          Email is invalid.
```
Comment (spadgett) [inline app/views/directives/create-secret.html]: Maybe

```html
Email must be in the form of <var>user@domain</var>.
```

File: app/views/directives/create-secret.html [hunk 2]
```diff
@@ -234,6 +234,11 @@
             required>
         </div>
       </div>
+      <div class="has-error" ng-show="secretForm.dockerEmail.$invalid && !secretForm.dockerEmail.$pristine">
```
Comment (spadgett) [inline app/views/directives/create-secret.html]: I'd check specifically for `secretForm.dockerEmail.$error.email` and have separate messages for invalid email values and missing required values.

**Comment by jhadvig** [comment]
@spadgett I've updated the PR based on your commets + added additional error messages for secret name, password & token, registry server address, registry user name, registry pw, registry email 

PTAL

File: app/views/directives/create-secret.html [hunk 3]
```diff
@@ -14,7 +14,7 @@
   <div ng-if="newSecret.type">
     <div class="form-group">
       <label for="secretName" class="required">Secret Name</label>
-        <span ng-class="{'has-error': nameTaken || (secretForm.secretName.$error.pattern && secretForm.secretName.$touched)}">
+        <span ng-class="{'has-error': nameTaken || ((secretForm.secretName.$error.required || secretForm.secretName.$error.pattern) && secretForm.secretName.$touched)}">
```
Comment (spadgett) [inline app/views/directives/create-secret.html]:
```html
<span ng-class="{'has-error': nameTaken || (secretForm.secretName.$invalid && secretForm.secretName.$touched)}">
```

File: app/views/directives/create-secret.html [hunk 4]
```diff
@@ -204,8 +214,13 @@
             required>
         </div>
       </div>
+      <div ng-show="secretForm.dockerUsername.$error.required && secretForm.dockerUsername.$touched" class="has-error">
+        <div class="help-block">
+          Email is required.
```
Comment (spadgett) [inline app/views/directives/create-secret.html]: "Username is required."

File: app/views/directives/create-secret.html [hunk 5]
```diff
@@ -234,6 +253,14 @@
             required>
         </div>
       </div>
+      <div class="has-error" ng-show="secretForm.dockerEmail.$invalid">
+        <div ng-show="secretForm.dockerEmail.$error.email" class="help-block">
```
Comment (spadgett) [inline app/views/directives/create-secret.html]:
```html
<div ng-show="secretForm.dockerEmail.$error.email && secretForm.dockerEmail.$touched" class="help-block">
```

That way the error doesn't flash as I type.

**Comment by jhadvig** [comment]
@spadgett thanks so much ! Commets addressed

**Comment by spadgett** [comment]
[merge]

**Comment by spadgett** [comment]
[merge]

**Pull request merged**